armi/nucDirectory/nuclideBases.py

```diff
@@ -143,625 +141,233 @@ def getAttributes(nuc):
     "TC": 99,
 }
 
-_riplEnvironVariable = "ARMI_RIPL_PATH"
-RIPL_PATH = None
 
+class NuclideInterface:
+    """An abstract nuclide implementation which defining various methods required for a nuclide object."""
 
-def isotopes(z):
-    return elements.byZ[z].nuclideBases
-
+    def getDatabaseName(self):
+        """Return the the nuclide label for the ARMI database (i.e. "nPu239")"""
+        raise NotImplementedError
```

Comment:
As a general comment, a lot of the machinery in this module could potentially be cleaned up using the `abc` package: https://docs.python.org/3/library/abc.html

Reply:
Good comment - I will consider this in the future.